src/react-components/src/components/01-atoms/Forms/DateRangeForm.tsx

```diff
@@ -0,0 +1,91 @@
+/* Input fields for date ranges. */
+import React from "react";
+import Button, { ButtonProps } from "../Button/Button";
+import HelperErrorText, { HelperErrorTextOptions } from "./HelperErrorText";
+import Input, { InputOptions } from "./Input";
+import { LabelOptions } from "./Label";
+import bem from "../../../utils/bem";
+
+export interface DateCheckFormProps {
+  /** Used for BEM css convention. */
+  blockName?: string;
+  /** Used for BEM css convention. */
+  baseClass?: string;
+
+  /** The Label to describe the form */
+  formLabel: JSX.Element;
+
+  /** The Label to describe the from input */
+  fromLabelOpts: LabelOptions;
+  /** The from input */
+  fromInputOpts: InputOptions;
+  /** Helper text for the from input */
+  fromHelper?: HelperErrorTextOptions;
+
+  /** The Label to describe the to input */
+  toLabelOpts: LabelOptions;
+  /** The to input */
+  toInputOpts: InputOptions;
+  /** Helper text for the to input */
+  toHelper?: HelperErrorTextOptions;
+
+  /** Whether the form is currently in an error state */
+  showError: boolean;
+  /** The error text to render */
+  error?: HelperErrorTextOptions;
+
+  /** The submit button */
+  buttonProps?: ButtonProps;
+}
+
+export default function DateRangeForm(props: DateCheckFormProps) {
+  const {
+    blockName = "date-range",
+    baseClass,
+    formLabel,
+    fromLabelOpts,
+    fromInputOpts,
+    fromHelper,
+    toLabelOpts,
+    toInputOpts,
+    toHelper,
+    showError,
+    error,
+    buttonProps } = props;
+    if (showError && !error) {
+      throw new Error("Error should be shown, but no Error Content exists");
+    }
```

Comment:
Nit, the `} = props;` should be on a new line and lines 55-57 are indented by 1 extra tab.